appgate/provider.go

```diff
@@ -261,3 +279,27 @@ func providerConfigure(ctx context.Context, d *schema.ResourceData) (interface{}
 
 	return c, diags
 }
+
+func defaultDeviceID() string {
+	readAndParseUUID := func() (uuid.UUID, error) {
+		// machine.ID() tries to read
+		// /etc/machine-id on Linux
+		// /etc/hostid on BSD
+		// ioreg -rd1 -c IOPlatformExpertDevice | grep IOPlatformUUID on OSX
+		// reg query HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Cryptography /v MachineGuid on Windows
+		// and tries to parse the value as a UUID
+		// https://github.com/denisbrodbeck/machineid
+		var id uuid.UUID
+		mid, err := machineid.ID()
+		if err != nil {
+			return id, err
+		}
+		return uuid.Parse(mid)
+	}
+	// if we can't get a valid UUID based on the machine ID, we will fallback to a random UUID value.
+	v, err := readAndParseUUID()
```

Comment:
Any reason why you declare a function for use just once?

Reply:
just for convenience, otherwise i need to do two  `if err != nil {}` statements.